flink-mesos/src/main/java/org/apache/flink/mesos/runtime/clusterframework/MesosResourceManager.java

```diff
@@ -312,6 +312,48 @@ private void recoverWorkers() throws Exception {
 	}
 
 	@Override
+	public void shutDown() throws Exception {
+		// shut down all components
+		if (taskMonitor != null) {
+			try {
+				actorSystem.stop(taskMonitor);
```

Comment:
We should try to gracefully stop the actors. I propose to use `Patterns.gracefulStop`, then collecting for all actors the returned futures, combining them and then wait on their completion with a timeout.